pallet-tfgrid/src/lib.rs

```diff
@@ -947,4 +959,36 @@ impl<T: Config> Module<T> {
 
         return ed25519_pubkey;
     }
+
+    pub fn fund_node_wallet(node_id: u32) {
+        if !Nodes::contains_key(&node_id) {
+            return
+        }
+        
+        let node = Nodes::get(node_id);
+        if !Farms::contains_key(node.farm_id) {
+            return
+        }
+        let farm = Farms::get(node.farm_id);
+
+        let node_twin = Twins::<T>::get(node.twin_id);
+        let farm_twin = Twins::<T>::get(farm.twin_id);
+        
+        let node_twin_balance: BalanceOf<T> = T::Currency::free_balance(&node_twin.account_id);
+        let minimal_balance = BalanceOf::<T>::saturated_from(200000 as u128);
+
+        if node_twin_balance <= minimal_balance {
+            let farmer_twin_balance: BalanceOf<T> = T::Currency::free_balance(&farm_twin.account_id);
+            let balance_to_transfer = BalanceOf::<T>::saturated_from(10000000 as u128);
+
+            if farmer_twin_balance <= balance_to_transfer {
+				debug::info!("farmer does not have enough balance to transfer");
+                return
+            }
+
+            debug::info!("Transfering: {:?} from farmer twin {:?} to node twin {:?}", &balance_to_transfer, &farm_twin.account_id, &node_twin.account_id);
+            T::Currency::transfer(&farm_twin.account_id, &node_twin.account_id, balance_to_transfer, KeepAlive)
+                .map_err(|_| debug::error!("Can't make transfer from farmer twin to node twin")).unwrap();
```

Comment:
map_err changes the type of the error, but the result will still be an error. Meaning, if transfer throws an error, the closure passed to `map_err` is invoked, changing the Err variant of the result to (). The unwrap call will then panic. Considering you don't care for the Ok result, the correct way to write this (as I assume you don't want to panic at all - this would kill the node) would be with an if let binding:

```rust
if let Err(_) = T::Currency::transfer(&farm_twin.account_id, &node_twin.account_id, balance_to_transfer, KeepAlive) {
    debug::error!("Can't make transfer from farmer twin to node twin");
};
```